### ISSUE TITLE
Optimize Experiments always fail if a user has inserted their UA snippet via Tag Manager

### ISSUE THREAD

**Issue by jamesozzie** [google/site-kit-wp#4084]
## Bug Description

As reported by [one user in the support forums](https://wordpress.org/support/topic/optimize-installation-not-found-analytics-tracking-code-not-found/page/2/) an Optimize experiment will always fail if a user has the Analytics, GTM and Optimize modules setup within Site Kit, with a Universal Analytics tag placed via Tag Manager. 

Rather than this being Site Kit related this may be as a result of a Optimize Chrome extension limitation (_an extension that is required for the "Check Installation" feature within Optimize_). While further testing is required this may be as a result of the UA tracking snippet not evident in a sites source code if placed via a Tag within Google Tag Manager.

## Steps to reproduce

1. Setup Analytics within Site Kit, and ensure all users are being tracked (including logged in users)
2. Setup an Experiment within Optimize, an AB test. 
3. Connect the Optimize module within Site Kit and connect the container just created 
3. Use the "Check installation" feature within Optimize - all should work fine
4. Connect GTM within Site Kit - to a container that has a Universal Type Analytics tag 
5. Site Kit will switch to the GTM placed snippet and the experiments check will fail

## Screenshots

![image](https://user-images.githubusercontent.com/41326532/134052881-645db2fc-07fc-462a-81da-548958811153.png)

![image](https://user-images.githubusercontent.com/41326532/134052939-2e04bf06-22ab-4df2-805f-05152e19adad.png)

## Additional Context

 - SK 1.41.0

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation Brief

* 

### Test Coverage

* 

## QA Brief

* 

## Changelog entry

*

**Comment by jamesozzie** [comment]
Further checks to be performed whether this "Check installation" feature stops an already existing experiment from running, I suspect this is not the case. The issue therefore is an experiment cannot be started - as this check is required to start an A/B test. 

If true one workaround for Site Kit users at present is to start the experiment by temporarily disconnecting the Tag Manager module within Site Kit and allowing the plugin to insert the Universal Analytics code. After successfully running a check within Optimize users can then switch to a GTM placed Analytics code by reconnecting GTM within the plugin.

**Comment by jamesozzie** [comment]
I performed [some additional testing](https://app.asana.com/0/1200206407034013/1201866554338978) on this today, in various scenarios. The outcome is that this does impact users of Tag Manager placed via Site Kit only. 
- **GTM and Optimize setup on a site without SK** - _no issues, optimize completes it's installation check successfully_
- **GTM setup manually and the Optimize module activated within SK (without the GTM and Analytics modules connected)** - _incomplete / not applicable_, as the Optimize module can't be setup without the Analytics module being connected
- **Analytics and Optimize modules connected, with GTM added manually (which includes a UA tag)** _Incomplete_, SK can't current recognize an Analytics property with a GTM tag if the GTM snippets were manually inserted  . See #4103
- **Analytics & GTM connected via Site Kit, Optimize manually added (ie. the module not connected)**: _[Analytics snippet not detected](https://recordit.co/8c9Jsn5moW)_
- **Analytics, GTM and Optimize connected via Site Kit**: _[Analytics snippet not detected](https://recordit.co/rqB0JfSAs5)_

So to summarize this issue remains, and is only applicable to sites that have the Analytics module connected via the Tag Manager module (_which includes a Universal - UA tag_).

Any change to #4103 may resolve this.

**Comment by bethanylang** [comment]
We haven't had any other reports of this and the issue is based on having UA active, which is going away in a few months.  :)  Closing.
